### PR TITLE
fix: focus management after navigation

### LINKED ISSUE #8438: Focus not reset when navigating away from page with `autofocus`
### Describe the bug

If a page has an `autofocus` element, navigating away from it will incorrectly focus the `autofocus` element on the origin page.

### Reproduction

[Stackblitz](https://stackblitz.com/edit/sveltejs-kit-template-default-yryvhn?file=src/routes/+page.svelte)

1. Click on "Page 1" link in nav. The input on the destination page is automatically focused since it has `autofocus`.
2. Click back to "Home". You'll see that the autofocus input is still the active element, even though it is not currently on the page. Depending on the browser and what is on the new page, your focus may start midway down the page when tabbing instead of starting at the top.

If you click between "Home" and "Page 2" instead (which does not have an `autofocus` attribute), focus is reset to the body as expected.

### Logs

_No response_

### System Info

```Shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 16.14.2 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 7.17.0 - /usr/local/bin/npm
  npmPackages:
    @sveltejs/adapter-auto: ^1.0.0 => 1.0.0 
    @sveltejs/kit: ^1.0.0 => 1.0.10 
    svelte: ^3.54.0 => 3.55.0 
    vite: ^4.0.0 => 4.0.4
```

### Severity

serious, but I can work around it

### Additional Information

_No response_

### LINKED ISSUE #8439: Cannot type in `<input>` after calling `.focus()` in `afterNavigate`
### Describe the bug

When an input is programmatically focused in `afterNavigate`, it becomes focused, but you can't type in it.

This may be due to resetting selection. I can repro in Chrome and Safari, but not Firefox.

You can work around the issue by wrapping the `focus` call in a `setTimeout`.

Related: calling focus in `afterNavigate` was suggested as a workaround in https://github.com/sveltejs/kit/issues/7895#issuecomment-1332968486

### Reproduction

[Stackblitz](https://stackblitz.com/edit/sveltejs-kit-template-default-jdtgye?file=src/routes/+layout.svelte)

Use Chrome or Safari.

1. Tab to the "Page 1" link and trigger it. The input on the page is programatically focused but you can't type in it. If you instead trigger the "focus thing" button (which calls the same function), the input is focused _and_ you can type in it
2. The same behavior occurs if you submit the form (which is a navigation) - the input is focused but you can't type in it.

### Logs

_No response_

### System Info

```Shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 16.14.2 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 7.17.0 - /usr/local/bin/npm
  npmPackages:
    @sveltejs/adapter-auto: ^1.0.0 => 1.0.0 
    @sveltejs/kit: ^1.0.0 => 1.0.10 
    svelte: ^3.54.0 => 3.55.0 
    vite: ^4.0.0 => 4.0.4
```

### Severity

serious, but I can work around it

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by dummdidumm**
- fixes #8438 by moving the refocus until after the rerender of the new page, so that old autofocus elements that will be removed are not focussed. It only calls focus if no manual focus by the user is detected.
- fixes #8439 by waiting for removeAllRanges() to be fired. This can't be tested through Playwright because there's no way to say "type something but don't focus the input before that"
- moves one test to deduplicate describe groups

In general I'd like to confirm that moving the focus management after `await tick()` doesn't break anything that isn't already covered by tests. I moved it and an additional check ensure we don't refocus when we shouldn't because someone already did it manually. Pinging @bluwy @Rich-Harris @benmccann @geoffrich because I think you all had to do with that part of the client already (if not, ignore this).

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

File: packages/kit/src/runtime/client/client.js [hunk 1]
```diff
@@ -325,13 +325,23 @@ export function create_client({ target, base }) {
...
 			await tick();
 
+			const activeElement = document.activeElement;
+			if (
+				!keepfocus &&
+				// reset focus only if any manual focus management didn't override it
+				(previousActiveElement === document.activeElement ||
+					// also refocus when activeElement is body already because the focus event might
+					// not have been fired on it yet
+					activeElement === document.body)
```
[the remark this answers is not in the document]
Comment (geoffrich) [reply packages/kit/src/runtime/client/client.js:340]: If I'm understanding correctly, this line is here to also reset focus when activeElement was set to `body` by the browser when the previously focused element was removed?

If so, I think we need it - in my [repro](https://stackblitz.com/edit/sveltejs-kit-template-default-yryvhn?file=src%2Froutes%2Fpage2%2F%2Bpage.svelte), even if the browser set activeElement to the `body` when the `<input>` is removed, you don't start tabbing from the top of the page; it starts from where the removed element was placed.

You can see this by adding an `<a href="whatever">` to the bottom of /page2. If you start on /page1, tab to the /page2 link and trigger it, and then start tabbing you'll notice that your focus skips over the navigation and starts at the link at the bottom of the page.

tl;dr good catch, I think we need it, it's weird though

**Review by Rich-Harris — APPROVED** [review]
